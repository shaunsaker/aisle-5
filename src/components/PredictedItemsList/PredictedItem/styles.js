import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
  },
  text: {
    flex: 1,
    ...styleConstants.fonts.types.paragraph,
    marginRight: styleConstants.dimensions.spacing.horizontal,
  },
  consumptionContainer: {
    flex: 2,
  },
  daysLeftTextContainer: {
    position: 'absolute',
    top: 0,
    right: styleConstants.dimensions.spacing.horizontal,
    bottom: 0,
    justifyContent: 'center',
    elevation: 100, // force it on top - android
  },
  daysLeftText: {
    ...styleConstants.fonts.types.small,
  },
  noDataText: {
    ...styleConstants.fonts.types.paragraph,
  },
});

export default styles;
