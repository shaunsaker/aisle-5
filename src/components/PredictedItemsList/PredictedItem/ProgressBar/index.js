import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import utils from '../../../../utils';
import styleConstants from '../../../../styleConstants';
import styles from './styles';

const propTypes = {
  value: PropTypes.number,
};

const defaultProps = {};

const ProgressBar = ({ value }) => {
  const valueAsInt = value / 100;
  const wrapperColor = utils.app.getColor(1 - valueAsInt);

  return (
    <View style={[styles.wrapper, { backgroundColor: wrapperColor }]}>
      <View style={[styles.container, { width: value }]} />
    </View>
  );
};

ProgressBar.propTypes = propTypes;
ProgressBar.defaultProps = defaultProps;

export default ProgressBar;
