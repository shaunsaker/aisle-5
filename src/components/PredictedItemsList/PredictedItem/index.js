import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import styles from './styles';

import ProgressBar from './ProgressBar';

const propTypes = {
  name: PropTypes.string,
  remainingQuantityPercentage: PropTypes.number,
  daysLeftUntilZeroQuantity: PropTypes.number,
  height: PropTypes.number,
};

const defaultProps = {};

const PROGRESS_BAR_WIDTH = 160;

const PredictedItem = ({
  name,
  remainingQuantityPercentage,
  daysLeftUntilZeroQuantity,
  height,
}) => {
  const daysLeftText = `${Math.floor(daysLeftUntilZeroQuantity)} days`;

  return (
    <View style={[styles.container, { height }]}>
      <Text numberOfLines={1} style={styles.text}>
        {name}
      </Text>

      <View style={styles.consumptionContainer}>
        <ProgressBar value={remainingQuantityPercentage} />

        <View style={styles.daysLeftTextContainer}>
          <Text style={styles.daysLeftText}>{daysLeftText}</Text>
        </View>
      </View>
    </View>
  );
};

PredictedItem.propTypes = propTypes;
PredictedItem.defaultProps = defaultProps;

export default PredictedItem;
