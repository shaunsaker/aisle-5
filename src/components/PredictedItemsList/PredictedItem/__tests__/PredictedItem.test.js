import React from 'react';
import renderer from 'react-test-renderer';

import PredictedItem from '..';
import PREDICTED_ITEMS_LIST from '../../../../mockData/predictedItemsList';

jest.mock('../ProgressBar', () => 'ProgressBar');

describe('PredictedItem', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<PredictedItem />);

      expect(component).toMatchSnapshot();
    });

    it('renders the not enough data state', () => {
      const predictedItem = PREDICTED_ITEMS_LIST[0];
      const component = renderer.create(<PredictedItem {...predictedItem} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the out of stock state', () => {
      const predictedItem = PREDICTED_ITEMS_LIST[1];
      const component = renderer.create(<PredictedItem {...predictedItem} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the default state', () => {
      const predictedItem = PREDICTED_ITEMS_LIST[2];
      const component = renderer.create(<PredictedItem {...predictedItem} />);

      expect(component).toMatchSnapshot();
    });
  });
});
