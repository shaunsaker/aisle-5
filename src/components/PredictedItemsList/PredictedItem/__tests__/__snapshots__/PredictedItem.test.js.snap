// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictedItem renders renders the default state 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "height": undefined,
          },
        ]
      }
    >
      <View
        style={Object {}}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontSize": undefined,
              },
            ]
          }
        >
          🍺
           
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#212121",
            "flex": 1,
            "fontFamily": "Muli-SemiBold",
            "fontSize": 17,
            "lineHeight": 25.5,
          }
        }
      >
        Beer
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <ProgressBar
      value={25}
    />
    <View
      style={
        Object {
          "bottom": 0,
          "elevation": 100,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#212121",
            "fontFamily": "Muli-SemiBold",
            "fontSize": 15,
          }
        }
      >
        2 days left
      </Text>
    </View>
  </View>
</View>
`;

exports[`PredictedItem renders renders the not enough data state 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "height": undefined,
          },
        ]
      }
    >
      <View
        style={Object {}}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#212121",
            "flex": 1,
            "fontFamily": "Muli-SemiBold",
            "fontSize": 17,
            "lineHeight": 25.5,
          }
        }
      >
        Whiskey
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontFamily": "Muli-SemiBold",
          "fontSize": 17,
          "lineHeight": 25.5,
        }
      }
    >
      Not enough data
    </Text>
  </View>
</View>
`;

exports[`PredictedItem renders renders the out of stock state 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "height": undefined,
          },
        ]
      }
    >
      <View
        style={Object {}}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontSize": undefined,
              },
            ]
          }
        >
          🍷
           
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#212121",
            "flex": 1,
            "fontFamily": "Muli-SemiBold",
            "fontSize": 17,
            "lineHeight": 25.5,
          }
        }
      >
        Wine
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <ProgressBar
      value={0}
    />
    <View
      style={
        Object {
          "bottom": 0,
          "elevation": 100,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#212121",
            "fontFamily": "Muli-SemiBold",
            "fontSize": 15,
          }
        }
      >
        0 days left
      </Text>
    </View>
  </View>
</View>
`;

exports[`PredictedItem renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "height": undefined,
          },
        ]
      }
    >
      <View
        style={Object {}}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#212121",
            "flex": 1,
            "fontFamily": "Muli-SemiBold",
            "fontSize": 17,
            "lineHeight": 25.5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <ProgressBar />
    <View
      style={
        Object {
          "bottom": 0,
          "elevation": 100,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#212121",
            "fontFamily": "Muli-SemiBold",
            "fontSize": 15,
          }
        }
      >
        NaN days left
      </Text>
    </View>
  </View>
</View>
`;
